### DISCUSSION TITLE
Why doesn't VERCEL_URL include the https prefix?

### DISCUSSION THREAD

**Discussion by tettoffensive** [vercel/vercel#5121] [Ideas]
I recently ran into an issue where I was using VERCEL_URL for firebase authentication "continue url". It was complaining until I realized that VERCEL_URL in my preview environment did not include the "https://" prefix. Once I manually prepended it, it worked.

I'm wondering if there's a reason why the url doesn't already include this prefix?

**Comment by paulogdm** [comment]
The URL should indeed include the protocol, in my opinion. However, it is unlikely this will be changed due to backward compatibility concerns, deprecation procedure and effort. It is just easier to include the protocol if you need to. Not every user is using the URL with the protocol explicitly added, for example.

**Reply by tettoffensive** [reply]
I see. Well I do believe it would be a better solution for people to explicitly remove the protocol when needed. But I understand the compatibility concerns. It's just kind of a "gotcha" that isn't mentioned in the docs

**Reply by stefanprobst** [reply]
@paulogdm would you be open to adding a new env variable which includes the protocol? currently `VERCEL_URL` cannot be used as a base url for the `URL` constructor, i.e. `new URL('/something', process.env.VERCEL_URL)` will throw.

**Reply by ericmand** [reply]
I like @stefanprobst's idea of adding a new environment variable called VERCEL_FULL_URL or something. I'm ran into this also.

**Comment by thexpand** [comment]
Hey Vercel team, is there any progress on this? Can't we just add another `VERCEL_FULL_URL` or `VERCEL_URL_WITH_PROTOCOL`, so that there is more flexibility for developers? Currently, this is a showstopper for me, because I want to directly re-assign an environment variable in my `.env` file like this on preview deployments on Vercel:
```
AUTH0_BASE_URL=$VERCEL_URL
NEXT_PUBLIC_AUTH0_BASE_URL=$VERCEL_URL
```

But that's not valid, because the `AUTH0_BASE_URL` is expected to include the protocol. See this issue for more info: https://github.com/auth0/nextjs-auth0/issues/417
